routable_doc.go

```diff
@@ -0,0 +1,9 @@
+package opensearchtools
+
+// RoutableDoc interface defines an OpenSearch document that can be routed to a specific index
+type RoutableDoc interface {
+	// GetID returns the document ID
+	GetID() string
+	// RouteToIndex returns the index the document should be routed to
+	RouteToIndex() string
```

Comment:
What do you all think of simply `Index` rather than `RouteToIndex`. Does the RouteTo bit tell you anything? It ultimately maps to simply `Index` in the official API OS client: https://github.com/CrowdStrike/opensearchtools/pull/2/commits/b82cd1a9688c8b26cd2a7341c5e3fdb8fa0f348d#diff-2d8ff542c31ebc376c35712f21a6edc13dff770edd0e2c611620ca3149b1bd90R31

Reply:
That's a good call. I can simplify the name.

Reply:
additional _idiomatic Go_ pedantry: should be `ID()` not `GetID()`